core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -592,6 +630,30 @@ public Builder localNodeId(String localNodeId) {
             return this;
         }
 
+        /**
+         * Checks that a node can be safely added to this node collection.
+         *
+         * @return null if all is OK or an error message explaining why a node can not be added.
+         *
+         * Note: if this method returns a non-null value, calling {@link #put(DiscoveryNode)} will fail with an
+         * exception
+         */
+        public String preflightPut(DiscoveryNode node) {
```

Comment:
`preflight` is the weirdest word and does not convey anything meaningful to me in the context of this class. Maybe call it `checkSafeAdd`?